packages/compiler/src/core/checker.ts

```diff
@@ -3105,6 +3076,15 @@ export function createChecker(program: Program): Checker {
       reportCheckerDiagnostic(createDiagnostic({ code: "spread-model", target: targetNode }));
       return [];
     }
+    if (parentModel.name === targetType.name) {
```

Comment:
```suggestion
    if (parentModel=== targetType) {
```

does this not work? with `.name` it could error for 2 same named model in different namespaces

Reply:
Updated.